draft-ietf-rats-msg-wrap.md

```diff
@@ -282,18 +289,30 @@ Since the collection type is recursive, implementations may limit the allowed de
 {: #fig-cddl-collection artwork-align="left"
    title="CDDL definition of the CMW collection format"}
 
-Although initially designed for the composite Attester use case, the CMW collection can be repurposed for other use cases requiring CMW aggregation.
+CMW itself provides no facilities for authenticity, integrity protection, privacy or attestation security.
+It is up to the designer of each use case to understand the security properties necessary and add them around the collection.
+A secure channel (e.g., using TLS) or some form of object security (e.g., using JWT) may be sufficient, but not always.
+
+When a CMW is used to carry the Evidence for composite or layered attestation for a single device, the security properties needed are that of attestation.
+In particular, all the members in a CMW must be bound together so that an attacker can not replace one Evidence message showing compromise with that from a non-compromised device.
+The authenticity and integrity protection MUST be attestation-oriented.
+For further security considerations about collections, see {{seccons-coll}}.
 
-A CMW collection carried in a `"cmw"` JWT claim ({{iana-jwt}}) MUST be a `json-collection`.
-A CMW collection carried in a `"cmw"` CWT claim ({{iana-cwt}}) MUST be a `cbor-collection`.
+### Relation to EAT `submods`
+
+EAT submods ({{Section 4.2.18 of -rats-eat}}) provide a facility for aggregating attestation that has built-in security and will be suitable for some of the same attestation Evidence use cases covered by CMW collections.
+However, compared to CMW collections, EAT submods are limited in two ways:
+
+1. Only EAT-formatted payloads are supported,
+2. Their top-down structure does not align well with the bottom-up approach layered attesters use to build the chain of trust, making them not ideal for modelling layered attestation.
```

Comment:
You can make an EAT submod a format other than EAT by defining a type for it. In CBOR-format EATs, that's just defining a new CBOR tag to identify it. Might be a bit weird to define a CBOR tag to identify an X.509 format attestation, but it is possible.

So I don't think 1. is true.

A more correct statement would be that EAT can't represent layered attestation when the topmost/outermost/lead attestation is not EAT format.

EAT seems fine for most layered attestation (see recent email to rats). Seems to me that CCA is doing rather unusual layered attestation. CCA attestation evidence spans devices, right? That's unusual for attestation IMO and beyond what I imagined anyone doing in EAT (and RATS).

EAT should not be presented as being unsuitable or bad at layered attestation.

Maybe say something like, while EAT submods are fine for most layered attestation, CMW is more flexible and can represent more complex or unusual layered attestation like a bottom-up chain of trust.

Reply:
> You can make an EAT submod a format other than EAT by defining a type for it. In CBOR-format EATs, that's just defining a new CBOR tag to identify it. Might be a bit weird to define a CBOR tag to identify an X.509 format attestation, but it is possible.
>
> So I don't think 1. is true.

This is not what I understand from reading  [4.2.18. submods (Submodules)](https://ietf-rats-wg.github.io/eat/draft-ietf-rats-eat.html#section-4.2.18).  There is no mention of the possibility of incorporating non-EAT tokens/claims-sets.  

Can you point me to the relevant text?

> A more correct statement would be that EAT can't represent layered attestation when the topmost/outermost/lead attestation is not EAT format.
> 
> EAT seems fine for most layered attestation (see recent email to rats). Seems to me that CCA is doing rather unusual layered attestation. CCA attestation evidence spans devices, right? That's unusual for attestation IMO and beyond what I imagined anyone doing in EAT (and RATS).
> 
> EAT should not be presented as being unsuitable or bad at layered attestation.
> 
> Maybe say something like, while EAT submods are fine for most layered attestation, CMW is more flexible and can represent more complex or unusual layered attestation like a bottom-up chain of trust.

We can certainly curate the text to avoid sounding adversarial vs EAT (that's certainly not my intent, on the contrary, I think EAT and CMW are an excellent pair.)

However, I stand by my assessment that submods are not great at layered attestation because they go counter the "natural" construction of the chain of trust of the device.

Typically (e.g., DICE, Arm CCA), the layer below signs the layer above, not the opposite.

Reply:
Re: incorporating non-EAT tokens/claims-sets, I think Laurence is referring to the fact that $EAT-CBOR-Tagged-Token is extensible. The potential for adding new types is noted in section 4.2.18 as follows:

```
The addition of further types is accomplished by augmenting the $EAT-CBOR-Tagged-Token socket or the $JSON-Selector socket.
```

You are correct the text does not explicitly call out that ASN.1 types, or other, could be accommodated this way but the means to support such is there.

Reply:
Thanks for the pointer!

Would it be fair to say something like:

> _"{{EAT}} allows the carrying of non-EAT-formatted types by augmenting the `$EAT-CBOR-Tagged-Token` socket or the `$JSON-Selector` socket. However, these need to be specified in subsequent standard documents updating the EAT specification."_

?

Reply:
I think so.

Reply:
Integrated in [39e1f53](https://github.com/ietf-rats-wg/draft-ietf-rats-msg-wrap/pull/117/commits/39e1f533af004805e30967fba3284101ac4c9a7f)

Reply:
Taking this a step further, we may as well drop this and add a new subsection _"Transporting CMW in EAT submods"_ that defines the sockets extensions:

```
$EAT-CBOR-Tagged-Token /= #6.TBD(cbor-CMW)
$JSON-Selector /= [type: "CMW", nested-token: json-CMW]
```

Where:

* `cbor-CMW` and `json-CMW` are defined in [Appendix A of CMW](https://www.ietf.org/archive/id/draft-ietf-rats-msg-wrap-08.html#appendix-A)
* `6.TBD` is the CBOR tag [requested for CMW](https://www.ietf.org/archive/id/draft-ietf-rats-msg-wrap-08.html#section-8.3)


@carl-wallace @laurencelundblade @gmandyam would that work?

Reply:
Adding use of submod extensibility here seems like a good idea to me.